docs/apps/advanced_app_features.rst

```diff
@@ -1,7 +1,88 @@
-Advanced App Features
-=====================
-
-See ``corehq.apps.app_manager.suite_xml.SuiteGenerator`` and ``corehq.apps.app_manager.xform.XForm`` for code.
+App Navigation Features
+=======================
+
+Navigation in CommCare is oriented around form entry. The goal of each CommCare session is to complete a form.
+Menus and case lists are tools for gathering the necessary data required to enter a particular form.
+App manager gives app builders direct control over many parts of their app's UI, but the exact sequence of screens
+a user sees is *indirectly* configured.
+
+Based on app configuration, HQ builds a ``suite.xml`` file that acts as a blueprint for the app. It outlines what
+forms are available to fill out, where they fit, and what data (like cases) they will need to function. CommCare
+interprets this configuration to decide which screens to show to the user and in what order.
+
+Much of the complexity of app manager code, and of building apps, comes from inferences HQ makes while building the
+suite file, especially around determining the set of data required for each form. These features that influence,
+but don't directly control, the suite also influence each other, in ways that may not be obvious. The following
+features are particularly prone to interact unexpectedly and should be tested together when any significant change
+is made to any of them:
+
+#. Display Only Forms
```

Comment:
This is the most checklist-y part. Open to feedback on whether or not these docs meet the spirit of the ticket.

Reply:
It's not quite what I had in mind but I think it's a good start

Reply:
The other specific addition I considered was laying out which features are known to be incompatible (basically looking at what situations throw app build validation errors). Do you think that'd be useful?